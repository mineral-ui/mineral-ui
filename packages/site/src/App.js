import React from 'react';
import BasicHelloExample
  from '@mineral-ui/hello/dist/es/__demo__/examples/Basic';

export default function App() {
  return (
    <div>
      <h1>Site</h1>
      <BasicHelloExample />
    </div>
  );
}
