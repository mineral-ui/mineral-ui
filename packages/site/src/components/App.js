/**
 * Copyright 2017 CA
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/* @flow */
import React from 'react';
import { Redirect, Route, Switch } from 'react-router-dom';
import { createStyledComponent } from '@mineral-ui/component-utils';
import createKeyMap from '../utils/createKeyMap';
import ComponentDoc from './ComponentDoc';
import Footer from './Footer';
import _Nav from './Nav';

type Props = {|
  className?: string,
  demos: Object | Array<Object>
|};

const styles = {
  app: ({ theme }) => ({
    fontFamily: theme.fontFamily_system,

    '@media(min-width: 45em)': {
      alignItems: 'stretch',
      display: 'flex',
      minHeight: '100vh'
    }
  }),
  nav: ({ theme }) => ({
    backgroundColor: theme.color_gray_10,
    border: `0 solid ${theme.borderColor}`,
    borderBottomWidth: '1px',

    '@media(min-width: 45em)': {
      borderBottomWidth: '0',
      borderRightWidth: '1px',
      width: `${16 * parseFloat(theme.spacing_double)}em`
    }
  }),
  main: {
    width: '100%'
  }
};

const Root = createStyledComponent('div', styles.app, {
  includeStyleReset: true
});
const Nav = createStyledComponent(_Nav, styles.nav);
const Main = createStyledComponent('main', styles.main);

const getDefaultDemo = (demos: Object): string => Object.keys(demos)[0];

export default function App({ className, demos }: Props) {
  if (!Array.isArray(demos) && demos.slug) {
    return <ComponentDoc {...demos} />;
  }

  const siteDemos = Array.isArray(demos) ? createKeyMap(demos, 'slug') : demos;

  return (
    <Root className={className}>
      <Nav demos={siteDemos} />
      <Main>
        <Switch>
          <Route
            path="/components/:componentId"
            render={route => {
              const componentId = route.match.params.componentId;
              // $FlowFixMe
              const selectedDemo = siteDemos[componentId];
              return <ComponentDoc {...selectedDemo} />;
            }}
          />
          <Redirect from="/" to={`/components/${getDefaultDemo(siteDemos)}`} />
        </Switch>
        <Footer />
      </Main>
    </Root>
  );
}
