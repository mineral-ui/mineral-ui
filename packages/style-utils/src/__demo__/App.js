/**
 * Copyright 2017 CA
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/* @flow */
import React from 'react';
import ComponentDoc from '../../../site/src/components/ComponentDoc';
import Default from './examples/Default';
import GlobalTheme from './examples/GlobalTheme';
import ComponentTheme from './examples/ComponentTheme';
import NestedGlobal from './examples/NestedGlobal';
import ShallowMerged from './examples/ShallowMerged';
import CreateThemedComponent from './examples/CreateThemedComponent';
import CreateStyledComponent from './examples/CreateStyledComponent';

const examples = [
  {
    title: 'Default',
    component: Default,
    description: 'Just a simple sample component to demonstrate the styling below.'
  },
  {
    title: 'Global theme var override',
    component: GlobalTheme,
    description: 'This will affect all components inside this ThemeProvider that use this variable.',
    source: `<ThemeProvider theme={{ color_primary: 'mediumvioletred' }}>
  <Sample />
</ThemeProvider>`
  },
  {
    title: 'Component theme var override',
    component: ComponentTheme,
    description: 'This will affect all components inside this ThemeProvider that use this variable, which should only be Sample components due to the namespace.',
    source: `<ThemeProvider theme={{ Sample_color: 'mediumvioletred' }}>
  <Sample />
</ThemeProvider>`
  },
  {
    title: 'Nested global theme var override',
    component: NestedGlobal,
    description: 'When a nested ThemeProvider provides the same variable, the merge works as expected and the most deeply-nested theme wins.',
    source: `<ThemeProvider theme={{ color_primary: 'cyan' }}>
  <ThemeProvider theme={{ color_primary: 'mediumvioletred' }}>
    <Sample />
  </ThemeProvider>
</ThemeProvider>`
  },
  {
    title: 'Shallow merged global theme var override',
    component: ShallowMerged,
    description: 'When a nested ThemeProvider provides a different variable, the merge works as expected.',
    source: `<ThemeProvider theme={{ color_primary: 'mediumvioletred' }}>
  <ThemeProvider theme={{ color_warning: 'tomato' }}>
    <Sample />
  </ThemeProvider>
</ThemeProvider>`
  },
  {
    title: 'Local theme var override via createThemedComponent',
    component: CreateThemedComponent,
    description: 'If you need to overwrite a theme variable, either global or component-specific, for a specific component.',
    source: `const MyThemedSample = createThemedComponent(Sample, {
  color_primary: 'mediumvioletred'
});

<MyThemedSample />`
  },
  {
    title: 'Style override via createStyledComponent',
    component: CreateStyledComponent,
    description: 'If you need to use completely custom styles (which can still reference the props & theme) on a component.',
    source: `const MyStyledSample = createStyledComponent(Sample, (props, theme) => ({
  outline: '3px dashed mediumvioletred',
  fontSize: theme.font_size_d
}));

<MyStyledSample />`
  }
];

const props = {
  description: 'Sample implentations of various methods of styling and style overrides.',
  examples,
  slug: 'style-utils',
  title: 'style-utils'
};

export default function App() {
  return <ComponentDoc {...props} />;
}
