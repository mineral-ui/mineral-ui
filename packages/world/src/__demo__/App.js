/**
 * Copyright 2017 CA
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import React from 'react';
import ComponentDoc from '../../../site/src/components/ComponentDoc';
import ComponentDocExample
  from '../../../site/src/components/ComponentDocExample';
import Default from './examples/Default';

export default function App() {
  const props = {
    description: "A simple component that renders the string, 'World'. It is primarily used for testing project configuration and package distribution.",
    slug: 'world',
    title: 'World'
  };

  return (
    <ComponentDoc {...props}>
      <ComponentDocExample>
        <Default />
      </ComponentDocExample>
    </ComponentDoc>
  );
}
