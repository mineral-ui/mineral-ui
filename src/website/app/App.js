/**
 * Copyright 2017 CA
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/* @flow */
import React, { Component } from 'react';
import { withRouter } from 'react-router';
import { Redirect, Route, Switch } from 'react-router-dom';
import { canUseDOM } from 'exenv';
import lighten from 'polished/lib/color/lighten';
import { pxToEm } from '../../styles';
import { mineralTheme, ThemeProvider } from '../../themes';
import BaselineGrid from './BaselineGrid';
import Router from './Router';
import siteColors from './siteColors';

declare var GOOGLE_TRACKING_ID: string;

type Props = {
  children?: any,
  className?: string,
  demoRoutes: Array<DemoRoute>,
  history: Object,
  location?: any
};

type DemoRoute = {
  description: string,
  slug: string,
  title: string
};

const siteTheme = {
  baseline_1: pxToEm(12),
  baseline_2: pxToEm(12 * 2),
  baseline_3: pxToEm(12 * 3),
  baseline_4: pxToEm(12 * 4),
  baseline_5: pxToEm(12 * 5),
  baseline_6: pxToEm(12 * 6),
  baseline_7: pxToEm(12 * 7),
  baseline_8: pxToEm(12 * 8),
  baseline_9: pxToEm(12 * 9),
  baseline_10: pxToEm(12 * 10),

  bp_moreSpacious: '@media(min-width: 48em)',

  bp_home_smallH3AndDown: '@media(max-width: 28.999em)',
  bp_home_bigH3: '@media(min-width: 29em)',
  bp_home_navCollapsedAndDown: '@media(max-width: 38.999em)',
  bp_home_navExpanded: '@media(min-width: 39em)',
  bp_home_getStartedLeftAlign: '@media(min-width: 43em)',
  bp_home_betweenMoreSpaciousAndGuidelinesMultiColumn:
    '@media(min-width: 48em) and (max-width: 60.999em)',
  bp_home_guidelinesMultiColumn: '@media(min-width: 61em)',

  bp_interior_bestPracticesMultiColumn: '@media(min-width: 61em)',

  maxContentWidth: '70em',
  maxTextWidth: '50em',
  navLink_color_active_narrow: lighten(0.12, siteColors.jade),
  sidebarWidth: pxToEm(200), // 180 + 20 for scrollbar
  textShadow: '2px 2px 2px rgba(0,0,0,0.2)',

  borderColor: siteColors.slate,
  borderColor_focus: siteColors.jade,
  color_text: mineralTheme.color_gray_80,
  color_text_primary: siteColors.jade,
  fontFamily: null,
  fontFamily_headline: `franklin-gothic-urw, ${mineralTheme.fontFamily_system}`,

  SectionPaddingHorizontal: pxToEm(30),
  SectionPaddingHorizontalWide: pxToEm(100),
  SectionPaddingVertical: pxToEm(12 * 6), // theme.baseline_6
  SectionPaddingVerticalWide: pxToEm(12 * 10), // theme.baseline_10

  SiteButton_backgroundColor_hover: mineralTheme.color_white,
  SiteButton_backgroundColor_primary: siteColors.jade,
  SiteButton_backgroundColor_primary_active: siteColors.jade_active,
  SiteButton_backgroundColor_primary_focus: siteColors.jade_focus,
  SiteButton_backgroundColor_primary_hover: siteColors.jade_hover,

  SiteHeading_color_2: siteColors.slateDarker,
  SiteHeading_color_3: siteColors.jade,
  SiteHeading_color_4: siteColors.slateDarker,
  SiteHeading_fontFamily: `franklin-gothic-urw, ${mineralTheme.fontFamily_system}`,
  SiteHeading_fontSize_1: pxToEm(40),
  SiteHeading_fontSize_1_wide: pxToEm(48),
  SiteHeading_fontSize_2: pxToEm(33),
  SiteHeading_fontSize_2_wide: pxToEm(40),
  SiteHeading_fontSize_3: pxToEm(23),
  SiteHeading_fontSize_3_wide: pxToEm(28),
  SiteHeading_fontSize_4: pxToEm(18),
  SiteHeading_fontSize_4_wide: pxToEm(22),
  SiteHeading_fontWeight_1: '300',
  SiteHeading_fontWeight_2: '300',
  SiteHeading_fontWeight_3: '300',
  SiteHeading_fontWeight_4: '500',
  SiteHeading_lineHeight_1: '1.1',
  SiteHeading_lineHeight_2: '1.2',
  SiteHeading_lineHeight_3: '1.5',
  SiteHeading_lineHeight_4: '1.5',

  SiteLink_borderColor_focus: siteColors.jade_focus,
  SiteLink_color: siteColors.jade,
  SiteLink_color_active: siteColors.jade_active,
  SiteLink_color_focus: siteColors.jade_focus,
  SiteLink_color_hover: siteColors.jade_hover
};

class App extends Component<Props> {
  constructor(props) {
    super(props);

    if (GOOGLE_TRACKING_ID) {
      // Analytics tracking of push state page views
      props.history.listen((location, action) => {
        if (canUseDOM && action === 'PUSH') {
          global.window.gtag('config', GOOGLE_TRACKING_ID, {
            page_path: location.pathname
          });
        }
      });
    }
  }

  componentDidUpdate(prevProps) {
    if (canUseDOM && this.props.location !== prevProps.location) {
      global.window.scrollTo(0, 0);
    }
  }

  render() {
    const { demoRoutes } = this.props;

    return (
      <ThemeProvider theme={siteTheme}>
        <div>
          <Switch>
            <Route
              exact
              strict
              path="/:url*"
              render={props => <Redirect to={`${props.location.pathname}/`} />}
            />
            <Route render={() => <Router demoRoutes={demoRoutes} />} />
          </Switch>
          <BaselineGrid />
        </div>
      </ThemeProvider>
    );
  }
}

export default withRouter(App);
