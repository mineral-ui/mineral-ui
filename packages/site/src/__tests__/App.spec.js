import React from 'react';
import { shallow } from 'enzyme';
import App from '../App';

function renderApp() {
  return shallow(<App />);
}

describe('App', () => {
  it('renders', () => {
    const app = renderApp();

    expect(app.exists()).toEqual(true);
  });

  it('renders correctly', () => {
    const app = renderApp();

    expect(app).toMatchSnapshot();
  });
});
