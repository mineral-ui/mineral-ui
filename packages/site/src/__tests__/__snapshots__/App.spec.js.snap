// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<div>
  <h1>
    Site
  </h1>
  <Basic />
</div>
`;
